src/main/java/ProjectOcean/Model/CoursePlanningSystem.java

```diff
@@ -1,156 +1,134 @@
 package ProjectOcean.Model;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Observable;
 
 /**
  * Represents the aggregate of the model
  */
 public class CoursePlanningSystem extends Observable {
 
 
-    private Student student;
-    private final Map<UUID, Course> courses;
+    private final Student student;
+    private final List<ICourse> courses;
 
+    /**
+     * Constructor call method generateCourses to instantiate a list of courses at startup of program.
+     * A student object is created.
+     */
     public CoursePlanningSystem() {
-        this.courses = generateCourses();
+        courses = generateCourses();
         this.student = new Student();
     }
 
     /**
      * @return returns all courses stored
      */
-    public Map<UUID, Course> getAllCourses() {
-        return Collections.unmodifiableMap(courses);
+    public List<ICourse> getAllCourses() {
+        return Collections.unmodifiableList(courses);
     }
 
     /**
      * Creates a list of hard coded courses
      * @return returns a list full of courses
      */
-    public Map<UUID, Course> generateCourses() {
-        Map courses = new HashMap<UUID, Course>();
-
-        Course course = new Course("DAT017","Maskinorienterad programmering", 7.5f, 1, "Roger Johansson", "Tenta/Laborationer", "Svenska", new ArrayList<>(), "www.google.com", "Lorem Ipsum");
-        courses.put(course.getId(), course);
-
-        course = new Course("EDA433","Grundläggande Datorteknik", 7.5f, 2, "Rolf Söderström", "Tenta/Laborationer", "Svenska", new ArrayList<>(), "www.google.com", "Lorem Ipsum");
-        courses.put(course.getId(), course);
+    public List<ICourse> generateCourses() {
 
-        course = new Course("MVE045","Matematisk Analys", 7.5f, 1, "Zoran Konkoli", "Tenta", "Svenska", new ArrayList<>(), "www.google.com", "Lorem Ipsum");
-        courses.put(course.getId(), course);
+        List<ICourse> courses = new ArrayList<>();
 
-        course = new Course("TMV206","Linjär Algebra", 7.5f, 3, "Lukás Malý", "Tenta", "Svenska", new ArrayList<>(), "www.google.com", "Lorem Ipsum");
-        courses.put(course.getId(), course);
+        CourseFactory.SetStudyPeriod(1);
+        CourseFactory.SetCourseInfo("DAT017","Maskinorienterad programmering", 7.5f);
+        CourseFactory.SetCourseDetails(new ArrayList<>(), "www.google.com", "Lorem Ipsum");
+        CourseFactory.SetCourseAccessibility("Rolf Söderström", "Tenta", "Svenska");
+        ICourse course = CourseFactory.CreateCourse();
```

Comment:
Would prefer all courses to be generated in order to see that everything works and for test purposes